File: app/pages/repos/_repo_id/models/_model_id/training_metrics/roc/props.ts
import { ModelLayoutInfo } from "layouts/model_layout"

export type Props = {
	aucRoc: number
	class: string

	modelId: string
	modelLayoutInfo: ModelLayoutInfo

	rocCurveData: Array<{
		falsePositiveRate: number
		truePositiveRate: number
	}>
	title: string
}

File: app/pages/repos/_repo_id/models/_model_id/training_metrics/roc/props.rs
use tangram_app_layouts::model_layout::ModelLayoutInfo;

#[derive(serde::Serialize)]
#[serde(rename_all = "camelCase")]
pub struct Props {
	pub id: String,
	pub roc_curve_data: Vec<ROCCurveData>,
	pub model_layout_info: ModelLayoutInfo,
	pub class: String,
	pub auc_roc: f32,
}

#[derive(serde::Serialize)]
#[serde(rename_all = "camelCase")]
pub struct ROCCurveData {
	pub false_positive_rate: f32,
	pub true_positive_rate: f32,
}
